flink-runtime/src/main/java/org/apache/flink/runtime/state/Snapshotable.java

```diff
@@ -54,5 +54,5 @@
 	 *
 	 * @param state the old state to restore.
 	 */
-	void restore(Collection<S> state) throws Exception;
+	void restore(R state) throws Exception;
```

Comment:
To me it is not really clear, why you've made this change here? Code wise it does not seem necessary.